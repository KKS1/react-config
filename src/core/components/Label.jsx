import React, { Component } from "react";

const Label = ({ schema, label, id }) => {
  return !schema && label ? <label htmlFor={id}>{label}</label> : "";
};

export default Label;
