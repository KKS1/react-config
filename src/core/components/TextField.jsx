import React, { Component } from "react";
import Label from "./Label";

export default class TextField extends Component {
  static defaultProps = {
    label: "Label",
    placeholder: "",
    name: "textField",
    onChange: () => {}
  };

  changeHandler = e => {
    this.props.onChange(e.target.value);
  };

  render() {
    const { placeholder, name, value, id, options } = this.props;
    const inputProps = options ? options.inputProps : {};
    return (
      <div className="c-label-field">
        <Label {...this.props} />
        <input
          id={id}
          placeholder={placeholder}
          type="text"
          name={name}
          value={value == null ? "" : value}
          {...inputProps}
          onChange={this.changeHandler}
        />
      </div>
    );
  }
}
