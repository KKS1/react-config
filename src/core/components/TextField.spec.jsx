import React, { Component } from "react";
import Label from "./Label";
import TextField from "./TextField";

describe("TextField", () => {
  test("render_TextField", () => {
    const wrapper = shallow(<TextField />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  test("attach_props", () => {
    const props = {
      id: "formOne_labelOne",
      value: "some text",
      options: {
        inputProps: {
          maxLength: 8
        }
      }
    };
    const wrapper = shallow(<TextField {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  test("call_onChange_prop", () => {
    const changeHandler = jest.fn();

    const props = {
      id: "formOne_labelOne",
      value: "some text",
      onChange: changeHandler
    };
    const wrapper = shallow(<TextField {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();

    instance.changeHandler({ target: { value: "changed text" } });

    expect(changeHandler).toHaveBeenCalledWith("changed text");
  });
});
