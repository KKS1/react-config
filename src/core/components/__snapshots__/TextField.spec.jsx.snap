// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField attach_props 1`] = `
TextField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "id": "formOne_labelOne",
    "label": "Label",
    "name": "textField",
    "onChange": [Function],
    "options": Object {
      "inputProps": Object {
        "maxLength": 8,
      },
    },
    "placeholder": "",
    "value": "some text",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <TextField
        id="formOne_labelOne"
        label="Label"
        name="textField"
        onChange={[Function]}
        options={
          Object {
            "inputProps": Object {
              "maxLength": 8,
            },
          }
        }
        placeholder=""
        value="some text"
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-label-field"
      >
        <Label
          id="formOne_labelOne"
          label="Label"
          name="textField"
          onChange={[Function]}
          options={
            Object {
              "inputProps": Object {
                "maxLength": 8,
              },
            }
          }
          placeholder=""
          value="some text"
        />
        <input
          id="formOne_labelOne"
          maxLength={8}
          name="textField"
          onChange={[Function]}
          placeholder=""
          type="text"
          value="some text"
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`TextField call_onChange_prop 1`] = `
TextField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "id": "formOne_labelOne",
    "label": "Label",
    "name": "textField",
    "onChange": [MockFunction],
    "placeholder": "",
    "value": "some text",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <TextField
        id="formOne_labelOne"
        label="Label"
        name="textField"
        onChange={[MockFunction]}
        placeholder=""
        value="some text"
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-label-field"
      >
        <Label
          id="formOne_labelOne"
          label="Label"
          name="textField"
          onChange={[MockFunction]}
          placeholder=""
          value="some text"
        />
        <input
          id="formOne_labelOne"
          name="textField"
          onChange={[Function]}
          placeholder=""
          type="text"
          value="some text"
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`TextField render_TextField 1`] = `
TextField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "label": "Label",
    "name": "textField",
    "onChange": [Function],
    "placeholder": "",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <TextField
        label="Label"
        name="textField"
        onChange={[Function]}
        placeholder=""
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-label-field"
      >
        <Label
          label="Label"
          name="textField"
          onChange={[Function]}
          placeholder=""
        />
        <input
          name="textField"
          onChange={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
